test/test_prototype_transforms.py

```diff
@@ -152,3 +152,49 @@ def test_normalize(self, transform, input):
     )
     def test_random_resized_crop(self, transform, input):
         transform(input)
+
+
+class TestRandomHorizontalFlip:
+    def input_tensor(self, dtype: torch.dtype = torch.float32) -> torch.Tensor:
+        return torch.tensor([[[0, 1], [0, 1]], [[1, 0], [1, 0]]], dtype=dtype)
+
+    def expected_tensor(self, dtype: torch.dtype = torch.float32) -> torch.Tensor:
+        return torch.tensor([[[1, 0], [1, 0]], [[0, 1], [0, 1]]], dtype=dtype)
+
+    def test_simple_tensor_p1(self):
+        input = self.input_tensor()
+
+        actual = transforms.RandomHorizontalFlip(p=1.0)(input)
+
+        assert torch.equal(self.expected_tensor(), actual)
```

Comment:
`torch.equal` will be deprecated by PyTorch core soon. For comparisons, one should use [`torch.testing.assert_close`](https://pytorch.org/docs/master/testing.html#torch.testing.assert_close). By default, it is an equality check for integer inputs and a closeness check for floating point inputs. If you want to enforce equality here (which is correct), you can either use `torch.testing.assert_close(..., atol=0, rtol=0)` or import `from common_utils import assert_equal` that does this for you.